tutorial.ipynb

```diff
@@ -490,20 +491,366 @@
     "\n",
     "keywords = Counter()\n",
     "for chunk in doc.noun_chunks:\n",
-    "    if nlp.vocab[chunk.lemma_].prob < - 8: # probablity value -8 is arbitraraly selected threshold\n",
+    "    if nlp.vocab[chunk.lemma_].prob < - 8: # probablity value -8 is arbitrarily selected threshold\n",
     "        keywords[chunk.lemma_] += 1\n",
     "\n",
     "keywords.most_common(20)"
    ]
   },
   {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "## Open task on the RAND Terrorism Dataset"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We have preprocessed the [RAND Terrorism Dataset](http://www.rand.org/nsrd/projects/terrorism-incidents.html) for this task. \n",
+    "\n",
+    "Can you find out the following using the code you have written?\n",
+    "\n",
+    "- Who are the persons mentioned in each article?\n",
+    "- What locations are mentioned in each article? Hint: a location just has a different label to a person\n",
+    "- From all of your entities, can you find out which named entities are terrorists from the syntactic relationships?\n",
+    "- With all of this information, can you plot a figure expressing the relationships between locations and terrorists?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "#Read file\n",
+    "terror_file = read_file('data/rand-terrorism-dataset.txt')\n",
+    "terror_doc = nlp(terror_file)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "#Get sentences\n",
+    "sentences = [s for s in terror_doc.sents]\n",
+    "\n",
+    "print(sentences[1:5])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "#Find persons and organisations mentioned in the text\n",
+    "\n",
+    "persons = Counter()\n",
+    "for ent in terror_doc.ents:\n",
+    "    if ent.label_ == 'PERSON' or  ent.label_ == 'ORG':\n",
+    "        persons[ent.lemma_] += 1\n",
+    "print(persons.most_common(30))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "#Find locations mentioned in the text\n",
+    "\n",
+    "locs = Counter()\n",
+    "for ent in terror_doc.ents:\n",
+    "    if ent.label_ == 'GPE':\n",
+    "        locs[ent.lemma_] += 1\n",
+    "print(locs.most_common(50))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "#You might think that in order to extract the terrorists from the text and ignore the civilians,\n",
```

Comment:
Can you add a space after # sign. Could you please apply these to all code the comments.